pennylane/pytrees.py

```diff
@@ -52,7 +104,115 @@ def register_pytree(pytree_type: type, flatten_fn: FlattenFn, unflatten_fn: Unfl
     Side Effects:
         ``pytree`` type becomes registered with available backends.
 
+    .. seealso:: :func:`~.flatten`, :func:`~.unflatten`.
+
     """
 
+    _register_pytree_with_pennylane(pytree_type, flatten_fn, unflatten_fn)
+
     if has_jax:
         _register_pytree_with_jax(pytree_type, flatten_fn, unflatten_fn)
+
+
+@dataclass(repr=False)
+class PyTreeStructure:
+    """A pytree data structure, holding the type, metadata, and child pytree structures.
+
+    >>> op = qml.adjoint(qml.RX(0.1, 0))
+    >>> data, structure = qml.pytrees.flatten(op)
+    >>> structure
+    PyTree(AdjointOperation, (), [PyTree(RX, (<Wires = [0]>, ()), [Leaf])])
+
+    A leaf is defined as just a ``PyTreeStructure`` with ``type=None``.
+    """
+
+    type: Optional[type] = None
+    """The type corresponding to the node. If ``None``, then the structure is a leaf."""
+
+    metadata: Metadata = ()
+    """Any metadata needed to reproduce the original object."""
+
+    children: list["PyTreeStructure"] = field(default_factory=list)
+    """The children of the pytree node.  Can be either other structures or terminal leaves."""
+
+    @property
+    def is_leaf(self) -> bool:
+        """Whether or not the structure is a leaf."""
+        return self.type is None
+
+    def __repr__(self):
+        if self.is_leaf:
+            return "PyTreeStructure()"
+        return f"PyTreeStructure({self.type.__name__}, {self.metadata}, {self.children})"
+
+    def __str__(self):
+        if self.is_leaf:
+            return "Leaf"
+        children_string = ", ".join(str(c) for c in self.children)
+        return f"PyTree({self.type.__name__}, {self.metadata}, [{children_string}])"
```

Comment:
```suggestion
        return f"PyTreeStructure({self.type.__name__}, {self.metadata}, [{children_string}])"
```

Reply:
Sticking with `Pytree` so that the str version can be a bit more terse. This leads to improve readability when the output starts to get long and complicated.